skrf/circuit.py

```diff
@@ -719,7 +749,10 @@ def s(self) -> np.ndarray:
         S : :class:`numpy.ndarray`
             global scattering parameters of the circuit.
         """
-        return self.X @ np.linalg.inv(np.identity(self.dim) - self.C @ self.X)
+        X = self.X
+        C = self.C
+        T = - C @ X + np.identity(self.dim)
```

Comment:
Isn't creating new intermediate variables consume more memory for nothing here?

Reply:
Sorry, after re-adopting the np.identity method, I forgot to remove the unnecessary intermediate variable. It has now been removed.